akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, StashBuffer, TimerScheduler }
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+
+/**
+ * INTERNAL API
+ *
+ * First (of four) behaviour of an PersistentBehaviour.
+ *
+ * Requests a permit to start recovering this actor; this is tone to avoid
+ * hammering the journal with too many concurrently recovering actors.
+ */
+@InternalApi
+private[akka] final class EventsourcedRequestingRecoveryPermit[Command, Event, State](
+  val persistenceId:    String,
+  override val context: ActorContext[Any],
+  override val timers:  TimerScheduler[Any],
+
+  val recovery: Recovery,
+
+  val callbacks: EventsourcedCallbacks[Command, Event, State],
+  val pluginIds: EventsourcedPluginIds
+) extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement {
+
+  import akka.actor.typed.scaladsl.adapter._
+
+  // has to be lazy, since we want to obtain the persistenceId
```

Comment:
this comment can't be right, `persistenceId` is a constructor parameter

Reply:
thanks